src/core/internal/array/casting.d

```diff
@@ -35,10 +35,12 @@ private void onArrayCastError()(string fromType, size_t fromSize, string toType,
     size_t index = 0;
     void add(const(char)[] m)
     {
+        import core.stdc.string : memcpy;
+
         auto N = msgLength - 1 - index;
         if (N > m.length)
             N = m.length;
-        msg[index .. index + N] = m[0 .. N];
+        memcpy(msg + index, m.ptr, N);
```

Comment:
Please put a comment noting this so it doesn't get reverted.

Reply:
Done.